tornado/web.py

```diff
@@ -2273,12 +2279,16 @@ class RedirectHandler(RequestHandler):
     .. versionchanged:: 4.5
        Added support for substitutions into the destination URL.
     """
-    def initialize(self, url, permanent=True):
+    def initialize(self, url, permanent=True, send_query=False):
         self._url = url
         self._permanent = permanent
+        self._send_query = send_query
 
     def get(self, *args):
-        self.redirect(self._url.format(*args), permanent=self._permanent)
+        to_url = self._url.format(*args)
+        if self._send_query and self.request.query:
+            to_url = to_url + '?' + self.request.query
```

Comment:
Use `tornado.httputil.url_concat` in case `self._url` already has some query parameters.